src/sidebar/comment.js

```diff
@@ -0,0 +1,130 @@
+import { useDataQuery, useDataMutation, useAlert } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    CircularLoader,
+    Button,
+    ButtonStrip,
+    ReactFinalForm,
+    TextAreaFieldFF,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useEffect, useState } from 'react'
+import styles from './comment.module.css'
+import LoadingError from './loading-error.js'
+import ToggleableUnit from './toggleable-unit.js'
+
+// TODO
+const query = {
+    comment: {
+        resource: 'comment',
+        id: ({ id }) => id,
+    },
+}
+
+// TODO
+const mutation = {}
+
+const Comment = ({ itemId }) => {
+    const [editing, setEditing] = useState(false)
+    const { called, loading, error, data, refetch } = useDataQuery(query, {
+        lazy: true,
+    })
+    const fetchComment = () => {
+        setEditing(false)
+        refetch({ id: itemId })
+    }
+    const errorAlert = useAlert(
+        i18n.t('There was a problem saving the data item comment.'),
+        { critical: true }
+    )
+    const [updateComment] = useDataMutation(mutation, {
+        onError: () => errorAlert.show(),
+        onComplete: fetchComment,
+    })
+
+    useEffect(fetchComment, [itemId])
+
+    if (!called || loading) {
+        return <CircularLoader small />
+    }
+
+    if (error) {
+        return (
+            <LoadingError
+                title={i18n.t(
+                    'There was a problem loading the comment for this data item'
+                )}
+            />
+        )
+    }
+
+    if (editing) {
+        return (
+            <ReactFinalForm.Form
+                onSubmit={({ comment }) =>
+                    updateComment({ id: itemId, comment })
+                }
+            >
+                {({ handleSubmit, submitting }) => (
+                    <form onSubmit={handleSubmit}>
+                        <ReactFinalForm.Field
+                            name="comment"
+                            component={TextAreaFieldFF}
+                            className={styles.textArea}
+                            initialValue={comment}
+                            dense
+                            autoGrow
+                        />
+                        <ButtonStrip>
+                            <Button
+                                small
+                                primary
+                                type="submit"
+                                loading={submitting}
+                            >
+                                {submitting
+                                    ? i18n.t('Saving...')
+                                    : i18n.t('Save comment')}
+                            </Button>
```

Comment:
Should this button be disabled while loading?

Reply:
Setting `loading=true` disables `@dhis2/ui` buttons: https://github.com/dhis2/ui/blob/afc6eb5ef84a9796330b7eeac8addd2dd337f786/components/button/src/button/button.js#L60